benchmarks/tpcc/src/main/java/com/google/cloud/pgadapter/tpcc/BenchmarkApplication.java

```diff
@@ -0,0 +1,246 @@
+import com.google.cloud.opentelemetry.metric.MetricConfiguration;
+import com.google.cloud.pgadapter.tpcc.config.PGAdapterConfiguration;
+import com.google.cloud.pgadapter.tpcc.config.SpannerConfiguration;
+import com.google.cloud.pgadapter.tpcc.config.TpccConfiguration;
+import com.google.cloud.pgadapter.tpcc.dataloader.DataLoadStatus;
+import com.google.cloud.pgadapter.tpcc.dataloader.DataLoader;
+import com.google.cloud.spanner.SessionPoolOptions;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.pgadapter.ProxyServer;
+import com.google.cloud.spanner.pgadapter.metadata.OptionsMetadata;
+import com.google.common.base.Stopwatch;
+import com.google.common.base.Strings;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdk;
+import io.opentelemetry.sdk.autoconfigure.AutoConfiguredOpenTelemetrySdkBuilder;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import io.opentelemetry.sdk.metrics.export.PeriodicMetricReader;
+import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class BenchmarkApplication implements CommandLineRunner {
+  private static final Logger LOG = LoggerFactory.getLogger(BenchmarkApplication.class);
+
+  public static void main(String[] args) {
+    try {
+      SpringApplication.run(BenchmarkApplication.class, args);
+    } catch (Throwable exception) {
+      exception.printStackTrace();
+    }
+  }
+
+  private final SpannerConfiguration spannerConfiguration;
+
+  private final PGAdapterConfiguration pgAdapterConfiguration;
+
+  private final TpccConfiguration tpccConfiguration;
+
+  public BenchmarkApplication(
+      SpannerConfiguration spannerConfiguration,
+      PGAdapterConfiguration pgAdapterConfiguration,
+      TpccConfiguration tpccConfiguration) {
+    this.spannerConfiguration = spannerConfiguration;
+    this.pgAdapterConfiguration = pgAdapterConfiguration;
+    this.tpccConfiguration = tpccConfiguration;
+  }
+
+  @Override
+  public void run(String... args) throws Exception {
+    ProxyServer server = pgAdapterConfiguration.isInProcess() ? startPGAdapter() : null;
+    String pgadapterConnectionUrl =
+        server == null
+            ? pgAdapterConfiguration.getConnectionUrl()
+            : String.format(
+                "jdbc:postgresql://localhost:%d/tpcc?preferQueryMode=simple",
```

Comment:
I don't think we should use `preferQueryMode=simple`. That will just slow things down, especially if the benchmark is modified to also use `java.sql.PreparedStatement`.

Reply:
Removed.